client/session.go

```diff
@@ -1429,17 +1526,12 @@ func (s *session) streamCollectedBlocksMetadata(
 		}
 
 		received, ok := metadata[m.id.Hash()]
-		if !ok {
+		if !ok || received.submitted {
 			received = &receivedBlocks{
 				results: make([]*blocksMetadata, 0, peersLen),
 			}
 			metadata[m.id.Hash()] = received
 		}
-		if received.submitted {
```

Comment:
hm..maybe i'm missing the point, what's the issue with leaving it where it is now?